### PR TITLE
chore: turn e2e enterprise tests into e2e premium tests

### PR DESCRIPTION
**Pull request by aslilac**
- Rename some e2e stuff from enterprise to premium
- Shuffle secrets around for a premium license
- Quit ignoring the results of the licensed e2e tests
- Fix some tests that weren't passing

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -499,19 +499,19 @@ jobs:
         working-directory: site
 
   test-e2e:
-    runs-on: ${{ github.repository_owner == 'coder' && (matrix.variant.enterprise && 'depot-ubuntu-22.04' || 'depot-ubuntu-22.04-4') || 'ubuntu-latest' }}
```
Comment (jaaydenh) [inline .github/workflows/ci.yaml:502]: just curious why this is not necessary anymore?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -535,38 +535,35 @@ jobs:
...
         working-directory: site
-        # Temporarily allow these to fail so that I can gather data about which
-        # tests are failing.
-        continue-on-error: true
 
       - name: Upload Playwright Failed Tests
         if: always() && github.actor != 'dependabot[bot]' && runner.os == 'Linux' && !github.event.pull_request.head.repo.fork
         uses: actions/upload-artifact@604373da6381bf24206979c74d06a550515601b9 # v4.4.1
         with:
-          name: failed-test-videos${{ matrix.variant.enterprise && '-enterprise' || '-agpl' }}
```
Comment (jaaydenh) [inline .github/workflows/ci.yaml:561]: do you know what the -agpl flag was for?

File: site/e2e/playwright.config.ts [hunk 3]
```diff
@@ -70,6 +70,11 @@ export default defineConfig({
 	],
 	reporter: [["./reporter.ts"]],
 	use: {
+		// It'd be very nice to add this, but there are some tests that need
+		// tweaking to make it work consistently (notably, ones that wait for agent
+		// stats on the workspace page. The default is like 50 seconds, which is
+		// way too long and makes it painful to wait for test runs in CI.
+		// actionTimeout: 5000, // 5 seconds
```
Comment (jaaydenh) [inline site/e2e/playwright.config.ts:77]: Would this be worth creating a separate issue to address this later?

File: site/e2e/helpers.ts [hunk 4]
```diff
@@ -191,31 +203,47 @@ export const createTemplate = async (
 		});
 	}
 
+	// If the organization picker is present on the page, select the default
+	// organization.
+	const orgPicker = page.getByLabel("Belongs to *");
```
Comment (jaaydenh) [inline site/e2e/helpers.ts:208]: This may need to be updated when the orgs redesign is implementred

**Review by jaaydenh — APPROVED** [review]
LGTM, just left a few questions.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (aslilac) [reply .github/workflows/ci.yaml:502]: the premium adds more tests, it just doesn't make sense to run it on a smaller, slower, less stable runner. idk why it was set up like this to begin with.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (aslilac) [reply .github/workflows/ci.yaml:561]: it's just a file name suffix, and calling it "-agpl" is _technically_ incorrect. it's an enterprise build, just unlicensed.

File: site/e2e/helpers.ts (hunk 4 above)
Comment (aslilac) [reply site/e2e/helpers.ts:208]: it's fine. there's tons of stuff like that.

**Pull request merged**